### PR TITLE
Reduxification User: get rid of isRtl method

### PR DESCRIPTION
**Pull request by lamosty**
Part of solving #19828.

In this PR, we replace all instances of `user.isRtl` method with `state/selectors/is-rtl` selector.

## Testing

Test commit by commit, smoke testing all the touched files/components.

### CODE REVIEW

File: client/extensions/woocommerce/components/compact-tinymce/index.js [hunk 1]
```diff
@@ -16,10 +16,10 @@ import 'tinymce/plugins/lists/plugin.js';
...
  */
 import i18n from 'components/tinymce/i18n';
-import userFactory from 'lib/user';
 import { wpautop } from 'lib/formatting';
 // TinyMCE plugins & dependencies
 import wplinkPlugin from 'components/tinymce/plugins/wplink/plugin';
+import { isRtl as isRtlSelector, getCurrentLocaleSlug } from 'state/selectors';
 
 class CompactTinyMCE extends Component {
 	static contextTypes = {
```
Comment (samouri) [inline client/extensions/woocommerce/components/compact-tinymce/index.js]: why are we using the context api instead of connecting?

**Comment by samouri** [comment]
Code LGTM, still need to test

**Comment by yoavf** [comment]
Seems like there's a regression with `AutoDirection` - which makes sure that content is using the correct text direction

My reader in `master`:
screen shot 2018-03-06 at 10 25 36

This branch:
screen shot 2018-03-06 at 10 26 52

cc @yurynix

**Comment by lamosty** [comment]
That's a great catch @yoavf! Should be fixed in becb0b4. Could you try again, please?

**Comment by yoavf** [comment]
@lamosty Looks good now 👍

**Pull request merged**